pkg/sysregistriesv2/system_registries_v2.go

```diff
@@ -570,6 +579,66 @@ func UnqualifiedSearchRegistries(ctx *types.SystemContext) ([]string, error) {
 	return config.UnqualifiedSearchRegistries, nil
 }
 
+// GetCredsStore return the list of credetntial store configed by creds_store for registries
+func GetCredsStore(ctx *types.SystemContext) ([]string, error) {
+	config, err := getConfig(ctx)
+	if err != nil {
+		return nil, err
+	}
+	return config.CredsStore, nil
+}
+
+// GetCredConfigs returns a slice of credential stores configred in registries.conf
+// caller should use GetRegistryCredentialHelpers to get cred helper of specified registry
+func GetCredConfigs(sys *types.SystemContext) ([]string, error) {
+	var (
+		credConfigs []string
+		credHelpers = make(map[string]bool)
+	)
+	config, err := getConfig(sys)
+	if err != nil {
+		return nil, errors.Wrapf(err, "unable to parse the registries.conf file")
+	}
+	for _, store := range config.CredsStore {
+		credHelpers[store] = true
+	}
+	registries, err := GetRegistries(sys)
```

Comment:
Just iterate `config.Registries` directly; this function already depends on the layout of `config`, and this way we avoid the lock in `getConfig`, don’t have to worry about using different snapshots of the configuration if it is read twice, and the case of `err != nil` vanishes.